src/provider/provider_tracking_windows.c

```diff
@@ -22,13 +24,17 @@ BOOL APIENTRY DllMain(HINSTANCE, DWORD fdwReason, LPVOID lpvReserved) {
     return TRUE;
 }
 #else
-typedef struct tracker_t {
-    tracker_t() { map = critnib_new(); }
-    ~tracker_t() { critnib_delete(map); }
-    critnib *map;
-} tracker_t;
-tracker_t TRACKER_INSTANCE;
-critnib *TRACKER = TRACKER_INSTANCE.map;
+INIT_ONCE init_once_flag = INIT_ONCE_STATIC_INIT;
+
+BOOL CALLBACK providerInit(PINIT_ONCE InitOnce, PVOID Parameter,
+                           PVOID *lpContext) {
+    TRACKER = critnib_new();
+    return TRACKER ? TRUE : FALSE;
+}
+
+void umfTrackingMemoryProviderInit(void) {
```

Comment:
It would be good to have a comment stating that we intentionally leak the tracker.